esmvalcore/preprocessor/_io.py

```diff
@@ -142,43 +142,73 @@ def _fix_cube_attributes(cubes):
         cube.attributes = attributes
 
 
-def concatenate(cubes):
-    """Concatenate all cubes after fixing metadata."""
-    _fix_cube_attributes(cubes)
-
-    concatenated = iris.cube.CubeList(cubes).concatenate()
-    if len(concatenated) == 2:
-        try:
-            concatenated[0].coord('time')
-            concatenated[1].coord('time')
-        except iris.exceptions.CoordinateNotFoundError:
-            pass
+def _by_two_concatenation(concatenated):
+    """Perform a by-2 concatenation to avoid gaps."""
+    try:
+        concatenated[0].coord('time')
+        concatenated[1].coord('time')
+    except iris.exceptions.CoordinateNotFoundError as exc:
+        msg = "One or both cubes {} is missing".format(concatenated) + \
+              " time coordinate: {}".format(str(exc))
+        raise ValueError(msg)
+    else:
+        concatenated = iris.cube.CubeList(concatenated).concatenate()
+        if len(concatenated) < 2:
+            return concatenated[0]
         else:
             concatenated = _concatenate_overlapping_cubes(concatenated)
+            if len(concatenated) == 2:
+                _get_concatenation_error(concatenated)
+            else:
+                return concatenated[0]
 
-    if len(concatenated) == 1:
-        cube = concatenated[0]
-        _fix_aux_factories(cube)
-        return cube
 
+def _get_concatenation_error(cubes):
+    """Raise an error for concatenation."""
     # Concatenation not successful -> retrieve exact error message
     try:
         iris.cube.CubeList(cubes).concatenate_cube()
     except iris.exceptions.ConcatenateError as exc:
         msg = str(exc)
     logger.error('Can not concatenate cubes into a single one: %s', msg)
     logger.error('Resulting cubes:')
-    for cube in concatenated:
+    for cube in cubes:
         logger.error(cube)
-        try:
-            time = cube.coord('time')
-        except iris.exceptions.CoordinateNotFoundError:
-            pass
-        else:
-            logger.error('From %s to %s', time.cell(0), time.cell(-1))
+        time = cube.coord("time")
+        logger.error('From %s to %s', time.cell(0), time.cell(-1))
+
     raise ValueError(f'Can not concatenate cubes: {msg}')
 
 
+def concatenate(cubes):
+    """Concatenate all cubes after fixing metadata."""
+    _fix_cube_attributes(cubes)
+
+    if len(cubes) > 2:
```

Comment:
```suggestion
    if len(cubes) > 1:
```
there is no need for a special case with only two cubes